src/snowflake/cli/plugins/nativeapp/codegen/snowpark/python_processor.py

```diff
@@ -299,3 +307,140 @@ def get_src_py_file_to_dest_py_file_map(
         # TODO: Does this case 5 need to be considered?
 
         return src_py_file_to_dest_py_file_map
+
+    def generate_sql_ddl_statements(
+        self, dest_file_py_file_to_collected_entities: Dict[Path, Optional[Any]]
+    ) -> Dict[Path, str]:
+        """
+        Generates SQL DDL statements based on the entities collected from a set of python files in the artifact_to_process.
+        """
+        dest_file_py_file_to_ddl_map: Dict[Path, str] = {}
+        for py_file in dest_file_py_file_to_collected_entities:
+
+            collected_entities = dest_file_py_file_to_collected_entities[
+                py_file
+            ]  # Collected entities is List[Dict[str, Any]]
+            if collected_entities is None:
+                continue
+
+            ddl_lst_per_ef: List[str] = []
+            for extension_function in collected_entities:
+                ddl_lst_per_ef.append(
+                    generate_create_sql_ddl_statements(extension_function)
+                )
+                ddl_lst_per_ef.append(
+                    generate_grant_sql_ddl_statements(extension_function)
+                )
+
+            dest_file_py_file_to_ddl_map[py_file] = "\n".join(ddl_lst_per_ef)
+
+        return dest_file_py_file_to_ddl_map
+
+
+def generate_create_sql_ddl_statements(ex_fn: Dict[str, Any]) -> str:
+    """
+    Generates a "CREATE FUNCTION/PROCEDURE ... " SQL DDL statement based on a dictionary of extension function properties.
+    Logic for this create statement has been lifted from snowflake-snowpark-python v1.15.0 package.
+    """
+
+    object_type = ex_fn["object_type"]
+    object_name = ex_fn["object_name"]
+
+    if object_type == "PROCEDURE" and ex_fn.get("anonymous", False):
```

Comment:
nit: why would anonymous need a default vs making your sanitization pass handle it?